src/openforms/submissions/form_logic.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict
+
+from json_logic import jsonLogic
+
+from ..forms.models.form import FormLogic
+from ..prefill import JSONObject
+from .models import SubmissionStep
+
+
+def set_property_value(
+    configuration: JSONObject,
+    component_key: str,
+    property_name: str,
+    property_value: str,
+) -> JSONObject:
+    for index, component in enumerate(configuration["components"]):
+        if "components" in component:
+            configuration = set_property_value(
+                component, component_key, property_name, property_value
+            )
+
+        if component["key"] == component_key:
+            configuration["components"][index][property_name] = property_value
+
+    return configuration
+
+
+def evaluate_form_logic(step: SubmissionStep, data: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Process all the form logic rules and mutate the step configuration if required.
+    """
+    # grab the configuration that can be **mutated**
+    configuration = step.form_step.form_definition.configuration
+
+    # ensure this function is idempotent
+    _evaluated = getattr(step, "_form_logic_evaluated", False)
+    if _evaluated:
+        return configuration
+
+    rules = FormLogic.objects.filter(form_step=step.form_step)
+    for rule in rules:
+        trigger = jsonLogic(rule.json_logic_trigger, data)
+        for action in rule.actions:
+            if action["type"] == "value":
+                configuration = set_property_value(
+                    configuration, rule.component, "value", trigger
+                )
```

Comment:
Decided to separate trigger & action - the value action itself can be specified with JSON logic but we'll decouple that from the trigger JSON logic.

Reply:
After discussion: we decided that also for the action type `value` the trigger should give a `true` or `false` result, so that it is consistent with the other action types.

The actions of type `value` will contain some jsonLogic field that can be used to extract the value of another component *if* the trigger evaluates to `true`.